src/lib/containers/MarkdownSynapse.tsx

```diff
@@ -0,0 +1,481 @@
+                try {
+                    goTo!.scrollIntoView({
+                        behavior: 'smooth',
+                        block: 'center',
+                        inline: 'center'
+                    });
+                } catch (e) {
+                    console.log('error on scroll', e);
+                }
+            } else if (event.currentTarget.id !== null) {
+                let idOfContent = anchor.getAttribute("data-anchor");
+                let goTo = this.markupRef.current!.querySelector(`#${idOfContent}`);
+                try {
+                    goTo!.scrollIntoView({
+                        behavior: 'smooth',
+                        block: 'center',
+                        inline: 'center'
+                    });
+                } catch (e) {
+                    console.log('error on scroll', e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Given input text, generate markdown object to be passed onto inner html of some container.
+     * @param {String} text The text being written in plain markdown
+     * @returns {Object} Dictionary to be passed into dangerouslySetInnerHTML with markdown text
+     */
+    createMarkup(text: string) {
+        let initText = this.state.md.render(text);
+        let cleanText = sanitizeHtml(initText, {
+            allowedTags: [
+                'span','code','h1','h2','h3','h4','h5','p','b','i',
+                'em','strong','a','id','table','tr','td','tbody','th',
+                'thead','button','div','image','ol','ul','li','svg','g','br','hr','summary',
+                'details'
+            ],
+            allowedAttributes: {
+                a: ['href'],
+                span: ['*'],
+                button: ['class'],
+                div: ['class', 'style'],
+                ul: ['class'],
+                ol: ['class'],
+                li: ['class'],
+                table: ['class'],
+                th: ['class'],
+                thead: ['class'],
+                h1: ['toc'],
+                h2: ['toc'],
+                h3: ['toc'],
+                h4: ['toc'],
+                h5: ['toc'],
+                h6: ['toc']
+            }
+        });
+        return { __html: cleanText };
+    }
+    /**
+     * Find all math identified elements of the form [id^=\"mathjax-\"]
+     * (e.g. <dom element id="mathjax-10"> text </dom element>)
+     * and transform them to their math markedown equivalents
+     */
+    processMath() {
+        if (!this.markupRef.current) {
+            return;
+        }
+        // use regex to grab all elements
+        let mathExpressions = this.markupRef.current.querySelectorAll('[id^="mathjax-"]');
+        // go through all obtained elements and transform them with katex
+        mathExpressions.forEach((element: any) => {
+            katex.render(element.textContent, element, {
+                // @ts-ignore: The docs for katex report conflicting information
```

Comment:
did you test math processing after ts conversion?

Reply:
will do that now

Reply:
confirmed that this will render math (had to add katex css files to the project), also confirmed that the type declarations are in fact inconsistent with the actual usage of katex so the ignore comment is necessary